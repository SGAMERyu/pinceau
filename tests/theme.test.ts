import { describe, expect, it } from 'vitest'

describe('@pinceau/theme', () => {
  it('should run', () => {
    expect(true).toBe(true)
  })
})
