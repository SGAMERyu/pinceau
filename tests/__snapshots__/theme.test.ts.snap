// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@pinceau/theme > formats/ > css.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "pinceau.css": "@media { :root {--pinceau-mq: initial; --primary: red; } }",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;

exports[`@pinceau/theme > formats/ > definitions.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "$pinceau/definitions": "export const definitions = {} as const",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;

exports[`@pinceau/theme > formats/ > hmr.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "$pinceau/hmr": "import { updateStyle } from '/@vite/client'


if (import.meta.hot) {
  import.meta.hot.on(
    'pinceau:theme',
    theme => {
      console.log({ theme })
      theme?.css && updateStyle('pinceau.css', theme.css)
    }
  )
}",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;

exports[`@pinceau/theme > formats/ > schema.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "$pinceau/schema": "export const schema = {
  \\"properties\\": {},
  \\"default\\": {}
} as const

export const GeneratedPinceauThemeSchema = typeof schema

export default schema",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;

exports[`@pinceau/theme > formats/ > typescript.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "$pinceau/theme": "export const theme = {
  \\"color\\": {
    \\"primary\\": {
      \\"value\\": \\"red\\"
    }
  }
} as const

export type GeneratedPinceauTheme = typeof theme

export type GeneratedPinceauMediaQueries = \\"dark\\" | \\"light\\" | \\"initial\\";

export type GeneratedPinceauPaths = \\"color.primary\\";

export default theme",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;

exports[`@pinceau/theme > formats/ > utils.ts > build format 1`] = `
{
  "buildDir": undefined,
  "outputs": {
    "$pinceau/utils": "export const utils = {}

export type GeneratedPinceauUtils = typeof utils

export default utils",
  },
  "theme": {
    "color": {
      "primary": {
        "value": "red",
        "variable": undefined,
      },
    },
  },
}
`;
